### DISCUSSION TITLE
504 Time out after creating token, works local but not on deployments

### DISCUSSION THREAD

**Discussion by nofurtherinformation** [nextauthjs/next-auth#4338] [Help]
Hi all. Many many thanks in advance for any suggestions you can provide here!

I'm building an app using NextAuth with an email provider (SMTP via AWS SES) with a postgres database backend. This works locally but times out (504 `FUNCTION_INVOCATION_TIMEOUT`) on Vercel. This happens after the database creates the verification token (looking manually at the tables shows it being created), but the function hangs and login times out. 

Here's my set up, mostly based on the examples for the email provider and Prisma adapter:

```javascript
// pages/api/auth/[...nextauth].js
import NextAuth from 'next-auth'
import EmailProvider from 'next-auth/providers/email'
import { PrismaAdapter } from '@next-auth/prisma-adapter'
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

const AuthOptions = {
	adapter: PrismaAdapter(prisma),
	providers: [
		EmailProvider({
			server: {
				host: process.env.EMAIL_SERVER_HOST,
				port: process.env.EMAIL_SERVER_PORT,
				auth: {
					user: process.env.EMAIL_SERVER_USER,
					pass: process.env.EMAIL_SERVER_PASSWORD
				}
			},
			from: process.env.EMAIL_FROM
		})
	],
	session: {
		jwt: true,
		maxAge: 7 * 24 * 60 * 60
	},
	jwt: {
		secret: process.env.JWT_SECRET,
		encryption: true
	},
	secret: process.env.SECRET,
	debug: true
}

export default NextAuth(AuthOptions)
```

Env variables, following examples, include: 
```
EMAIL_FROM
EMAIL_SERVER_PORT
EMAIL_SERVER_HOST
EMAIL_SERVER_PASSWORD
EMAIL_SERVER_USER
DATABASE_URL
// and on prod
SECRET
JWT_SECRET
NEXTAUTH_URL
```

And my database setup uses the Schema from the [Prisma adapter tutorial](https://next-auth.js.org/adapters/prisma), and appears correct.

Here's what I've tried so far, unsuccessfully: 
- Confirm database is open to all traffic (0.0.0.0) for inbound and outbound traffic
- Tried a different database cloud provider (GCP instead of AWS) - same issue of local working, hangs on Vercel
- Swapping in a simple credentials provider to test short term, still no luck
- Swapping in FaunaDB instead of Postgres/Prisma, similar issue where a token is generated in `verification_tokens` but then hangs
- Hosting on netlify

This is all good on local, but confusingly doesn't work when deployed. Any help is much appreciated here, thanks very much for reading and your time!

**Comment by ajhenry** [comment]
@nofurtherinformation Were you able to find a solution? I’m also running into this

**Reply by ajhenry** [reply]
For anyone else running into this, this turned out to be an issue with `@mui/material-icons` package.

It’s too big to import so the `/api/*` routes fail to load.

https://github.com/vercel/next.js/issues/44321

**Comment by jinsley8** [comment]
I had this issue with Heroku Postgres free tier. I just tested Supabase Postgres and it's working fine, likely because the database responds faster.

You can try to migrate your db to a test DB in Supabase and see if that works too. If it does then it's likely the DB response time. I also read that upgrading your Vercel account allows a longer timeout.

https://supabase.com/docs/guides/migrations/heroku

**Comment by SadmanYasar** [comment]
I was facing this exact problem in a Strapi and Nextjs project. I fixed it by removing the trailing "/" in NEXTAUTH_URL eg. ```https://example.com``` instead of ``https://example.com/``. Then make sure to redeploy in Vercel.

**Reply by jjalonso** [reply]
it didnt solve my issue, tried using it like you said in prodand didnt work, also without NEXTAUTH_URL as doc say...

**Comment by jjalonso** [comment]
Not my case.
